.github/workflows/CI.yml

```diff
@@ -156,8 +156,8 @@ jobs:
           docker exec $(docker ps --filter "name=migration-console" -q) pipenv run pytest /root/lib/integ_test/integ_test/replayer_tests.py --unique_id="testindex" -s
       - name: Clean up migrations docker images before caching
         run: |
-          docker stop $(docker ps -q) && docker rm $(docker ps -aq)
-          docker image ls --format '{{.Repository}}:{{.Tag}}' | grep '^migrations/' | grep -v '<none>' | xargs -I {} docker image rm {}
+          docker stop $(docker ps -q) && docker system prune --volumes -f
+          docker image ls --format '{{.Repository}}:{{.Tag}}' | grep '^migrations/' | xargs -I {} docker image rm {}
```

Comment:
why are you deleting migrations images?  They might build up, but that's true for most things being cached at any level.  We already have largely disjoint containers.  A migration console change should be able to test w/ RFS, replayer, otel, grafana containers and not need to rebuild them, right?

Reply:
Migration image deletion has already been in the cache.

The reason is two-fold:
1. Our cache hashing logic is only on the dockerfiles and thus does not take into account the source files, and thus only supports the base images.
2. The current cache action doesn't support partial hits, we either have a hit and pull the cache and don't publish back to the cache, or we miss and repopulate the cache afterwards.

Reply:
Building off the points @AndreKurait already made:

1) The reason we are getting rate limited is from downloading from the docker repo servers.  The benefit to this workflow is caching our dependencies - not the images built in this repro.
2) GitHub has limits on space usage of the actions cache - if we are not caching migration images we are saving space.